### PR TITLE
Clarification of involvements

### LINKED ISSUE #220: Clarify the involvements section (1)
During the review of #205 there were some comments regarding the definitions and explanations use in the `involvements` property:

---

@tolim stated in https://github.com/oasis-tcs/csaf/pull/205#discussion_r611769150:

> It is unclear to me how this value should be used. I would assume that `disputed` in the context of the party means, that the named party and its affiliation to the vulnerability is disputed (and not the vulnerability report in its entirety).
> If there is no security implication, the vulnerability CVSS score would indicate this by a value of 0.
> Is there any case, where the value `disputed` would be used? At least from a vendor perspective, this value will never be useful as disputed or non-responding parties will not be mentioned in the advisory.

---

@sthagen replied in https://github.com/oasis-tcs/csaf/pull/205#discussion_r611778803:

> Defining categories in deeply nested structures sometimes makes us miss the forrest for the trees - I will read CVRF v1.2 again to see how we did there. Thanks for bringing this scope / use question up @tolim

---

This issue is used to track the progress and provide a place for discussions.

### PR DESCRIPTION
**Pull request by tschmidtb51**
- resolves #220 and #221
- add property `date`
- correct use of `vendor`: replace by `party` the involvement should always be seen as tuple => document producer has received `status` from `party` on `date`
- adapt description to be able to disclose the coordination timeline
- add tests (#195)

### CODE REVIEW

**Comment by tschmidtb51** [comment]
@tolim: Please review as you opened the initial issues.
@zmanion: Please review as you have a unique perspective regarding the coordinator side.

**Comment by tschmidtb51** [comment]
We will keep this as a draft PR until, we received the reviews.

**Review by sthagen — APPROVED** [review]
LGTM already.  Esp. The optional date information may open the door for controversial statements being able to coney a real ongoing discussion in a single CSAF document.

Yesterday party x was involved and summary says this, today same party contributes another summary :wink: … let us see, what the other members make out of it.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 1]
```diff
@@ -2079,23 +2079,28 @@ List of involvements (`involvements`) of value type `array` with 1 or more items
     },
 ```
 
-Every Involvement item of value type `object` with the 2 mandatory properties Party (`party`), Status (`status`) and the optional property Summary (`summary`) is a container, that allows the document producers to comment on their level of Involvement (or engagement) in the vulnerability identification, scoping, and remediation process.
+Every Involvement item of value type `object` with the 2 mandatory properties Party (`party`), Status (`status`) and the 2 optional properties Date of involvement (`date`) and Summary (`summary`) is a container, hat allows the document producers to comment on the level of involvement (or engagement) of themselves (or third parties) in the vulnerability identification, scoping, and remediation process. It can also be used to disclose the coordination timeline. The ordered tuple of the values of `party`, `status` and `date` (if present) SHALL be unique within `involvements`.
```
Comment (zmanion) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:2082]: s/hat/that
And the comma before "hat" is perhaps strictly not necessary but defer to editor/author.

**Comment by zmanion** [comment]
OK with me, would want to/will test (in production) and provide any further feedback. Noted a small typo.

**Review by santosomar — APPROVED** [review]

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 1 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md]: Thank you.

**Comment by santosomar** [comment]
This issue was discussed during the CSAF TC Monthly meeting on May 26th, 2021. The changes suggested were approved.

**Comment by tschmidtb51** [comment]
Merged into the `oasis-tcs/csaf:master` through #265.

Resolve #220, #221.

**Pull request merged**